scripts/FormatSource.ps1

```diff
@@ -0,0 +1,290 @@
+<#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for license information.
+
+this file work is a derivative of https://github.com/microsoft/MixedReality-GraphicsTools-Unreal/blob/main/Tools/scripts/Common.psm1
+#>
+
+<#
+.SYNOPSIS
+    Run clang-format on all source files.
+.PARAMETER Path
+    Path to the directory (or file) to be processed recursively. By default scans the entire repo.
+.PARAMETER ClangFormat
+    Path to clang-format executable, e.g. "C:\Tools\clang-format.exe"
+.PARAMETER ModifiedOnly
+    Scan only files modified in current git checkout.
+.PARAMETER Staged
+    Check only files staged for commit
+.PARAMETER ChangesFile
+    Scan only files listed in provided txt file (one path per line). Paths need to be relative to repo root.
+.PARAMETER Verify
+    Whether to fail if files are not formatted (instead of applying changes).
+.PARAMETER UseVS2019
+    Use ClangFormat provided with Visual Studio 2019. This version will take precedence over -ClangFormat parameter
+    and PATH variable.
+.PARAMETER NoFail`
+    Do not set RC=1 when errors found, i.e. only report errors in output.
+#>
+[CmdletBinding()]
+param (
+    [string]$Path = $null,
+    [string]$ClangFormat = $null,
+    [boolean]$ModifiedOnly = $True,
+    [boolean]$Staged = $false,
+    [string]$ChangesFile = $null,
+    [boolean]$Verify = $false,
+    [boolean]$UseVS2019 = $true,
+    [boolean]$NoFail = $false
+)
+
+# Only check source files
+$FilePatterns = "\.(h|cpp|hpp|c)$"
+
+#PerfApp is C++/Cx project and not supported
+$IgnoreFolders = "(Generated Files|PerfApp|android|ios|nodejs|vscode|community|dotnet|node_modules|out|.git|.vs|.vscode|bin|CMakeFiles|generated|debug|x64)$"
```

Comment:
This feels like it should be read from a file rather than hardcoded in the script...

Reply:
keep it simple in this PR. so no change will be made